onnx/version_converter/BaseConverter.h

```diff
@@ -0,0 +1,78 @@
+// Version converter interface for ONNX models between different opset versions.
+
+#pragma once
+
+#include "onnx/common/ir.h"
+#include "onnx/common/ir_pb_converter.h"
+#include "onnx/common/stl_backports.h"
+#include "onnx/proto_utils.h"
+#include "onnx/defs/schema.h"
+#include <utility>
+#include <iostream>
+#include <stdlib.h>
+// TODO: Remove this import once actual adapters are imported
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+// TODO: Consider creating interface for this class.
+class BaseVersionConverter {
+  // Schema for adapters: {<op_name>:{<from_domain>$<from_version>:{<to_domain>
+  // <to_version>: adapter}}}
+  protected:
+    std::unordered_map<std::string, std::unordered_map<std::string, std::unordered_map<std::string, std::unique_ptr<Adapter>>>> adapters;
+
+  public:
+    BaseVersionConverter() = default;
+
+    virtual ~BaseVersionConverter() = default;
+
+    // adapter_lookup should be called in convert_version when the user would
+    // like to identify the proper registered adapter in the adapters map for
+    // a given Node from a certain version to another. It should only be called
+    // when the user knows that an adapter should exist for the given context.
+    const Adapter& adapter_lookup(const Node* op,
+        const OpSetID& initial_version,
+        const OpSetID& target_version) const {
+      const std::string& op_name = op->kind().toString();
+      const std::string& initial = initial_version.toString();
+      const std::string& target = target_version.toString();
+      // Find appropriate adapter in adapters map for provided initial and target versions
+      // TODO: Consider abstracting elements of this that are specific to
+      // DefaultConverter to separate methods here and maintain the procedure in Base Converter
+      const auto& op_adapters = adapters.find(op_name);
```

Comment:
`adapters.find()` returns an iterator object. This is basically just a pointer and cheap to copy, so you don't need a reference here. Also, the reference here actually breaks your code, because (as in the previous comment), the iterator object gets destroyed if you don't store it and your reference will point to something dead.